mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -212,20 +213,32 @@ def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_p
             )
 
             if failed_downloads:
```

Comment:
We might want to throw immediately if the size of `failed_downloads` is large. Let's say you have 100 chunks to upload, and 90 of them fail, we throw.

Reply:
This might be a bit tricky to get the logic just right. If we do percentage basis, 2 failures out of 3 total would be the same as 200 failures out of 300 total (we would probably want to retry the first one, but not the 2nd one). If we do a scalar cutoff of say 50 (purely arbitrary for argument's sake), 49 failures out of 51 total is probably something we wouldn't want to retry, but failing 51 out of 5000 is probably something worth retrying. 
I'm worried about how complicated this logic might need to be to capture refined retry logic.

Do you think for now we could just brute-force retry all failures and revisit more advanced logic if we find it is an issue for users later?

Reply:
revisit later sounds good.